tests/test_unit.py

```diff
@@ -109,6 +109,13 @@ def testFilesBeneathRecursive(self):
             self.assertIn(r".\d\4.txt", files)
             self.assertIn(r".\d\e\5.txt", files)
 
+    def testCaseInsensitiveReplace(self):
+        replace = lambda s: clcache.caseInsensitiveReplace(s, b"PaTtErN", b"replacement")
+        self.assertEqual(replace(b"pattern may appear"), b"replacement may appear")
+        self.assertEqual(replace(b"infix paTTErn embed"), b"infix replacement embed")
+        self.assertEqual(replace(b"Pattern and PATTERN and pattern"), b"replacement and replacement and replacement")
+        self.assertEqual(replace(b"and finally PATTERN"), b"and finally replacement")
+
```

Comment:
I suppose that in case you decide to drop the custom `caseInsensitiveReplace` function in favor of `re.sub`, you could also drop this test.